packages/flutter/lib/src/cupertino/activity_indicator.dart

```diff
@@ -42,6 +46,14 @@ class CupertinoActivityIndicator extends StatefulWidget {
   /// Defaults to 10px. Must be positive and cannot be null.
   final double radius;
 
+  /// Determines the number of spinner segments that will be show. Typical usage would
+  /// display all segments, however, this allows for more fine-grained control such as
+  /// during pull-to-refresh when the drag-down action shows one segment at a time as
+  /// the user continues to drag down.
+  ///
+  /// Defaults to 1.0. Must be between 0.0 and 1.0 inclusive, and cannot be null.
+  final double progress;
```

Comment:
+1. This is implicitly controlling whether it's static or animating. It should definitely be pointed out in the docs. Perhaps allow null and make it mutually exclusive with animating? Since it's a bit harder to explain that 1 specific value of this double has a secondary meaning. 

I also strongly encourage consistency with https://api.flutter.dev/flutter/material/ProgressIndicator/value.html

Reply:
I wonder if it is worth creating an explicit constructor for the "progressive release" version? That might help with trying to keep the mutually exclusive values separate... not sure if there's any specific guidelines around when we should/shouldn't create a separate constructor?

Reply:
I think that's a great idea! Many widgets have more than one constructor so it doesn't look like a problem to me.

Reply:
Have added a new constructor called `CupertinoActivityIndicator.partiallyRevealed()` but am open to suggestions about better naming (I'm not super-thrilled about what I have now). Also updated some of the internal comments to be more accurate.